icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -209,12 +209,16 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         }
 
         int correctedNumBreaks = 0;
+        int previous = -1;
         for (int i = numBreaks - 1; i >= 0; i--) {
             int pos = charPositions[t_boundary[i]] + startPos;
-            if (!(foundBreaks.contains(pos) || pos == startPos)) {
-                foundBreaks.push(charPositions[t_boundary[i]] + startPos);
-                correctedNumBreaks++;
+            if (pos > previous) {
```

Comment:
if (pos > previous && pos != startPos) {

Reply:
why is the java and c++ logic different?

Reply:
In line-214, pos may have duplicate value.
C++ uses "if (utextPos > prevUTextPos)" to filter the duplicate ones while Java uses  foundBreaks.contains(pos)
Make the implementation synchronous between C++ and Java.

Reply:
Resolved

Reply:
BTW, since this is really just part of ICU-21699 and your ICU-21699 is not done yet, I think we should just use that bug number instead. You really do not need to file another bug at this point